### ISSUE TITLE
There should be no default page transitions for desktop and web platforms.

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter#99052]
As page transitions are not common on desktop and web applications, the default transitions on these platforms should be removed.

An attempt at this was tried with #82596, but it had an issue with the [delay was still occurring](https://github.com/flutter/flutter/issues/88221) even though the transition gone.

**Comment by johnpryan** [comment]
Right now users can use [a custom Page object that disables the transition](https://github.com/flutter/packages/blob/main/packages/go_router/lib/src/custom_transition_page.dart#L160), but requires some custom code to switch based on the platform. Also, there [needs to be a 1ms delay](https://github.com/csells/go_router/issues/205) when this is used with go_router. 

Instead, does it make sense to define a new PageTransitionsBuilder?

```
class NoAnimationPageTransitionsBuilder extends PageTransitionsBuilder {
  const NoAnimationPageTransitionsBuilder();

  @override
  Widget buildTransitions<T>(
      PageRoute<T> route,
      BuildContext context,
      Animation<double> animation,
      Animation<double> secondaryAnimation,
      Widget child,
      ) {
    return child;
  }
}

```

And then use that as the default for linux, macOS, and Windows in the `PageTransitionsTheme` class?

```
  static const Map<TargetPlatform, PageTransitionsBuilder> _defaultBuilders = <TargetPlatform, PageTransitionsBuilder>{
    TargetPlatform.android: FadeUpwardsPageTransitionsBuilder(),
    TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
    TargetPlatform.linux: FadeUpwardsPageTransitionsBuilder(), // Should be NoAnimationPageTransitionsBuilder?
    TargetPlatform.macOS: CupertinoPageTransitionsBuilder(), // Should be NoAnimationPageTransitionsBuilder?
    TargetPlatform.windows: FadeUpwardsPageTransitionsBuilder(), // Should be NoAnimationPageTransitionsBuilder?
  };
```

Am I understanding correctly that the only other thing that's needed is a way to customize the transition duration based on the platform?

**Comment by Braj65** [comment]
Does this also looks related? - #41564

**Comment by willlockwood** [comment]
> Also, #95764 seems related

Yes, that was my initial thought as well. I think likely the same fix would solve both of these. Basically, the solution would be that the `PageTransitionsTheme` should in some way be able to specify durations (as described in a very basic sense on that issue).

If no one is currently working this, I can try to figure that out 👍🏻

> Does this also looks related? - #41564

The behavior they're complaining about is the same, but that seems to be more focused on the fact that `TargetPlatform` isn't able to specify transitions for web? I'd imagine that as probably a completely different change, although I can't say for sure without an idea of how `TargetPlatform` will get assigned when `kIsWeb` is true

**Comment by jacobsimionato** [comment]
I think this should be higher than P5, because clients are having to work around this issue currently to meet user expectations. Also, we're looking at changing the default behavior which could be surprising / disruptive to some apps, and so the sooner we do it, the less apps will be disrupted.

**Comment by tmpfs** [comment]
I agree with @jacobsimionato this should be higher priority, for those coming across this issue here is a workaround:

```dart
class NoPageTransitionsBuilder extends PageTransitionsBuilder {
  const NoPageTransitionsBuilder();

  @override
  Widget buildTransitions<T>(
    PageRoute<T>? route,
    BuildContext? context,
    Animation<double> animation,
    Animation<double>? secondaryAnimation,
    Widget child,
  ) {
    return child;
  }
}
```

And then in `ThemeData` configure the `pageTransitionsTheme`:

```dart
    pageTransitionsTheme: const PageTransitionsTheme(
      builders: <TargetPlatform, PageTransitionsBuilder>{
        TargetPlatform.android: FadeUpwardsPageTransitionsBuilder(),
        TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
        TargetPlatform.linux: NoPageTransitionsBuilder(),
        TargetPlatform.macOS: NoPageTransitionsBuilder(),
        TargetPlatform.windows: NoPageTransitionsBuilder(),
      },
    ),
```

**Comment by rydmike** [comment]
@jacobsimionato I'm getting and access denied error when trying to read the  internal Google document. Is it somehing you can share with at least read access?


Looks very interesting and relevant to somehting I'm working on.

**Comment by benhawkinsicon** [comment]
Am I correct in saying **_there is no way to change default page transition durations in Flutter?_** 

The workaround @tmpfs posted doesn't remove the secondaryAnimation duration, so while it fixes navigating to a page, it does not fix navigating back. Therefore the only way to make web routing behave correctly is to wrap every individual route in a `NoTransitionPage`?

I would agree this should be a higher priority, as the only workaround requires extra code on every route, which will require quite a lot of refactoring across **all** Flutter-web projects once the fix rolls around.

**Comment by benhawkinsicon** [comment]
Example:

Using @tmpfs example yields a back delay:

https://github.com/flutter/flutter/assets/105246345/af92588d-9b1f-49fe-afd6-7f983bce6069


Only possible to remove by wrapping in `NoTransitionPage`:

https://github.com/flutter/flutter/assets/105246345/92d04ef0-70fe-42a0-b1bf-26bdccf80efe

So we actually shouldn't even touch `PageTransitionsTheme` as `NoTransitionPage` on every route makes it redundant.

**Comment by tmpfs** [comment]
@benhawkinsicon , yes that is annoying, there are other issues about the delay problem:

https://github.com/flutter/flutter/issues/27777
https://github.com/flutter/flutter/issues/88221

I really wish this was fixed properly in the Flutter framework but for now all I have is this workaround.

**Comment by danwild** [comment]
+1

I'm 2 days into building my first app in flutter and I was **immediately confused** by why my large web pages routes were awkwardly animating.

To make things worse (i.e. more complex..), I want to make my app _adaptive_.

In the adaptive paradigm, I would argue that:

1. animations for a web app running on a LARGE screen (i.e. desktop browser) make _no_ sense; but
2. animations for a web app on a SMALL screen make _perfect_ sense.. (e.g. a Drawer widget animating the slide open/close on a mobile browser).

The above workaround (thanks @tmpfs ! I'll use this pattern until something better presents) seems to be _platform_ specific (i.e. blanket case for all web screens) and doesn't cover my use-case, which I suspect would be a common one.

Following 👀

**Comment by tmpfs** [comment]
@darrenaustin is there an idea of when this might land please? I find this delay on back navigation personally very frustrating as I navigate the app (300ms is very noticeable) and I am sure my users will find the delay perplexing.

If there is something I can do to help move this forward please let me know.

**Comment by eEQK** [comment]
This can be worked around by specifying  `PageRoute.reverseTransitionDuration`

`go_router` example, but `CustomTransitionPage` is from SDK:
```dart
  @override
  Page<void> buildPage(BuildContext context, GoRouterState state) =>
      CustomTransitionPage(
        child: const FooScreen(),
        key: state.pageKey,
        transitionDuration: Duration.zero,
        reverseTransitionDuration: Duration.zero,
        transitionsBuilder: (_, __, ___, child) => child,
      );
```

more practical example:
```dart
@immutable
class HomeRoute extends GoRouteData {
  const HomeRoute();

  @override
  Page<void> buildPage(BuildContext context, GoRouterState state) =>
      _NoTransition(state, child: const HomeScreen());
}


class _NoTransition extends CustomTransitionPage {
  _NoTransition(GoRouterState state, {required super.child})
      : super(
          key: state.pageKey,
          transitionsBuilder: (_, __, ___, child) => child,
          reverseTransitionDuration: Duration.zero,
          transitionDuration: Duration.zero,
        );
}
```
```

**Comment by tmpfs** [comment]
@eEQK , this isn't working for me.

I added this to a `pageBuilder` and I still see the delay:

```dart
class _NoTransition extends CustomTransitionPage {
  _NoTransition(GoRouterState state, {required super.child})
      : super(
          key: state.pageKey,
          transitionsBuilder: (_, __, ___, child) => child,
          reverseTransitionDuration: Duration.zero,
          transitionDuration: Duration.zero,
        );
}

// Standard page builder.
Page<dynamic> pageBuilder(
    BuildContext context, GoRouterState state, Widget child) {
  
  if (isDesktop()) {
    return _NoTransition(state, child: child);
  }

  return MaterialPage(child: child);
}
```

Then I assign the `pageBuilder` to the route and still the standard 300ms delay. Any ideas?

**Comment by eEQK** [comment]
Not sure how I can help you @tmpfs since I created a minimal example and it works:
https://dartpad.dev/?id=9c8f887f324018bceb11c7c7406c0ba3
